code/game/objects/structures/crates_lockers/closets/__closet.dm

```diff
@@ -83,10 +89,15 @@ var/global/list/closets = list()
 
 /obj/structure/closet/proc/can_open()
 	if((setup & CLOSET_HAS_LOCK) && locked)
-		return 0
+		return FALSE
 	if((setup & CLOSET_CAN_BE_WELDED) && welded)
-		return 0
-	return 1
+		return FALSE
+	if(lock)
+		if(usr)
+			try_unlock(usr, usr.get_active_held_item())
```

Comment:
Not sure if I'm a huge fan of using `usr` here, would prefer passing `mob/user` from `toggle()` -> `open()` -> `can_open()`. Is there a reason we can't do that?